src/plugins/data_table/filter_view.js

```diff
@@ -206,35 +257,23 @@ var DataTableFilter = (function(options) {
     },
 
     _onToggleClick: function (event) {
-      var table = this.table;
       event.stopImmediatePropagation();
-      var currentFilterMenu = $('.filterMenu', this.$el);
-      if ((table.activeFilterMenu) && (table.activeFilterMenu.is(currentFilterMenu))) {
-        table.activeFilterMenu.slideUp(100);
-        table.activeFilterMenu = null;
-      } else if (table.activeFilterMenu) {
-        table.activeFilterMenu.slideUp(100, function () {
-          table.activeFilterMenu = currentFilterMenu;
-          table.activeFilterMenu.slideDown(200);
-        });
-      } else {
-        table.activeFilterMenu = currentFilterMenu;
-        table.activeFilterMenu.slideDown(200);
-      }
     },
 
     _onFilterClick: function () {
       $("input[type=text]", this.head).trigger("change");
       this.table.dataTable._fnAjaxUpdate();
+      this.$(".toggle-filter-button").popoverMenu('hide');
     },
 
     _onClearClick: function () {
-      $("input[type=text]", this.head).val("");
-      $("input[type=checkbox]", this.head).attr("checked", false);
-      $("input", this.head).trigger("change");
+      $("input[type=text]", this.child("filter-menu").$el).val("");
+      $("input[type=checkbox]", this.child("filter-menu").$el).attr("checked", false);
+      $("input", this.child("filter-menu").$el).trigger("change");
```

Comment:
hmm seems a better encapsulation would be for this base class to have the buttons rendered (as alluded to above in DRYing up the code), and keep these click handlers, but this onClearClick could emit an event that the child views listened for, and each was in charge of clearing their own data.  Rather than reaching into these filter-menus